src/lsp/cobol_preproc/compdir_grammar.mly

```diff
@@ -30,10 +30,11 @@
 %token NE              "<>"         [@keyword (* symbol *) "<>"]
 
 %token CDIR_DEFINE     [@keyword ">>DEFINE", "$DEFINE"]
-%token CDIR_ELIF       [@keyword ">>ELIF", ">>ELSE-IF"]
-%token CDIR_ELSE       [@keyword ">>ELSE"]
-%token CDIR_END_IF     [@keyword ">>END-IF"]
-%token CDIR_IF         [@keyword ">>IF"]
+%token CDIR_ELIF       [@keyword ">>ELIF", "$ELIF", ">>ELSE-IF", "$ELSE-IF"]
+%token CDIR_ELSE       [@keyword ">>ELSE", "$ELSE"]
+%token CDIR_END        [@keyword "$END"]       (* Note: no `>>END` equivalent *)
+%token CDIR_END_IF     [@keyword ">>END-IF", "$END-IF"]
```

Comment:
Note: `$END-IF` _does_ work and is found in existing code-bases, but is undocumented from MF (it is documented as `$END`. You may want to note that so that any later adjustment does not remove that.

Reply:
Thanks for the notice; I'll add a comment on that.